lib/screens/recruit/search/project_search.dart

```diff
@@ -0,0 +1,108 @@
+import 'package:flutter/material.dart';
+import 'package:guam_community_client/commons/guam_progress_indicator.dart';
+import 'package:guam_community_client/providers/recruit/recruit.dart';
+import 'package:guam_community_client/screens/recruit/search/filter_button.dart';
+import 'package:guam_community_client/screens/recruit/search/filter_status.dart';
+import 'package:provider/provider.dart';
+
+import '../../../providers/user_auth/authenticate.dart';
+import '../../../styles/colors.dart';
+import '../../search/search_app_bar.dart';
+import 'project_search_body.dart';
+import 'project_search_text_field.dart';
+
+
+
+class ProjectSearch extends StatefulWidget {
+  @override
+  State<ProjectSearch> createState() => _ProjectSearchState();
+}
+
+class _ProjectSearchState extends State<ProjectSearch> {
+  final Map filter = new Map();
+  late FocusNode myFocusNode;
+  bool cancel = true;
+
+  void cancelSearch(bool) {
+    setState(() => cancel = bool);
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    myFocusNode = FocusNode();
+  }
+
+  @override
+  void dispose() {
+    myFocusNode.dispose();
+    super.dispose();
+  }
+
+
+  @override
+  Widget build(BuildContext context) {
+    final recruitProvider = context.watch<Recruit>();
+
+    return Scaffold(
+      backgroundColor: GuamColorFamily.purpleLight3,
+      appBar: SearchAppBar(
+        title: ProjectSearchTextField(filter: filter, focusNode: myFocusNode, cancelSearch: cancelSearch),
+        trailing: FilterButton(filter: filter, setSkill: (skill){
+          setState(() {
+            if(skill == null){
+              filter.remove('skill');
+            }else{
+              filter['skill'] = skill;
+            }
+          });
```

Comment:
아래처럼 setSkill, setPosition, setDue의 값(함수 type)을 모듈화하여 사용도 가능할 것 같습니다! 
(e는 toString 필요 없어도 될 것 같은데 확인해봐야겠네요)
```
_setFilter(e) {
  setState(() {
    String eString = e.toString;
    if (e == null) {
      filter.remove(eString);
    } else {
      filter[eString] = e;
    }
  });
}
```

```suggestion
        trailing: FilterButton(filter: filter, setSkill: (skill) => _setFilter(skill),
```